File: ArasControl.Application/DTOs/AnimalDto.cs
using ArasControl.Domain.Entities.Enum;

namespace ArasControl.Application.DTOs
{
    public class AnimalDto
    {
        public string Name { get; set; }
        public string Breed { get; set; }
        public string Color { get; set; }
        public double WeightKg { get; set; }
        public double HeightCm { get; set; }
        public Guid HarasId { get; set; }
        public Guid OwnerId { get; set; }
        public DateTime DateOfBirth { get; set; }
        public Gender Sex { get; set; }
        public string MicrochipId { get; set; }
        public string RegistrationNumber { get; set; }
        public string Temperament { get; set; }
        public string MedicalHistory { get; set; }
    }

}

File: ArasControl.WebApp/arascontrol-frontend/src/app/animals/models/animal-dto.ts

export interface AnimalDto {
  id: string;
  name: string;
  breed: string;
  color: string;
  dimensions?: {
    weight: number;
    height: number;
  }; 
  harasId: string;
  ownerId: string;
  dateOfBirth: string;
  sex: 'Male' | 'Female';
  microchipId?: string;
  registrationNumber?: string;
  temperament?: string;
  medicalHistory?: string;
}
